### DISCUSSION TITLE
Tutorial gradCAM 3D seems to have inverted heatmap

### DISCUSSION THREAD

**Discussion by CamilleDelgrange** [Project-MONAI/tutorials#1797] [General]
Bug - GradCAM: 
https://github.com/Project-MONAI/tutorials/blob/main/modules/interpretability/class_lung_lesion.ipynb

Following this tutorial, and comparing with the original paper on GradCAM (Grad-CAM: Visual Explanations from Deep Networks via Gradient-Based Localization), I have the impression that there is an issue somewhere with the activation score or the heatmap colors because red should be related to high importance score and blue the reverse. 

<img width="431" alt="image" src="https://github.com/user-attachments/assets/46e44376-c0da-498e-b655-a37ede371d37">

How can I fix that in my situation ?

**Comment by KumoLiu** [comment]
Hi @CamilleDelgrange, thanks for your interest here. 
Yes, the `default_normalizer` used in GradCam will flip magnitudes.
https://github.com/Project-MONAI/MONAI/blob/dev/monai/visualize/class_activation_maps.py#L37

You can simply modify it by adding a customized normalizer.
```
def custom_normalizer(x):
    def _compute(data):
        scaler = ScaleIntensity(minv=0.0, maxv=1.0)
        return np.stack([scaler(i) for i in data], axis=0)
    if isinstance(x, torch.Tensor):
        return torch.as_tensor(_compute(x.detach().cpu().numpy()), device=x.device)
    return _compute(x)

monai.visualize.GradCAM(nn_module=model_3d, target_layers="class_layers.relu", postprocessing=custom_normalizer)
```

Hope it helps, thanks.
